packages/sails-ng-common/test/unit/validators.test.ts

```diff
@@ -283,6 +283,163 @@ describe("Validator", async () => {
                 },
                 // jsonata-expression - expression evaluation is not available in sails-ng-common
                 // This is tested in the sails tests.
+                {
+                    title: "orcid - expect pass (valid with hyphens)",
+                    args: {
+                        value: "0000-0002-1825-0097", definition: formValidatorsSharedDefinitions,
+                        block: {class: "orcid"},
+                    },
+                    expected: null,
+                },
+                {
+                    title: "orcid - expect pass (valid without hyphens)",
+                    args: {
+                        value: "0000000218250097", definition: formValidatorsSharedDefinitions,
+                        block: {class: "orcid"},
+                    },
+                    expected: null,
+                },
+                {
+                    title: "orcid - expect pass (valid with X checksum)",
+                    args: {
+                        value: "0000-0002-1694-233X", definition: formValidatorsSharedDefinitions,
+                        block: {class: "orcid"},
+                    },
+                    expected: null,
+                },
+                {
+                    title: "orcid - expect failure (URL https)",
+                    args: {
+                        value: "https://orcid.org/0000-0002-1825-0097", definition: formValidatorsSharedDefinitions,
```

Comment:
Are urls ever valid ORCIDs? If not, then I think the 4 url tests are excessive - only one or two are needed.

Reply:
Had them there because originally the code was allowing the full URL. The problem with that is that the format is inconsistent and makes using the ORCID more complicated. Now the code just allows formats that if you prepend `https://orcid.org/<number>`, the URL resolves.